src/mtype.d

```diff
@@ -7536,12 +7536,21 @@ public:
         }
         if (exp.op == TOKscope)
         {
-            ScopeDsymbol sds = (cast(ScopeExp)exp).sds;
+            auto sds = (cast(ScopeExp)exp).sds;
             if (sds.isPackage())
             {
                 error(loc, "%s has no type", exp.toChars());
                 goto Lerr;
             }
+            if (auto ti = sds.isTemplateInstance())
+            {
+                if (ti.needsTypeInference(sc))
+                {
+                    // Bugzilla 15550: ti is a partial instantiattion form fun!tiargs.
```

Comment:
Should be `instantiation`.

And would it be helpful to put some of this comment in the error message?  Maybe `expression (%s) is a partially instantiated template and has no type`?